enterprise/users/src/main/java/io/crate/metadata/UsersPrivilegesMetaData.java

```diff
@@ -74,11 +74,11 @@ public static UsersPrivilegesMetaData copyOf(@Nullable UsersPrivilegesMetaData o
                 String ident = privilege.ident();
                 assert ident != null : "ident must not be null for privilege class 'TABLE'";
                 if (ident.equals(sourceIdent)) {
-                    privileges.add(new Privilege(privilege.state(), privilege.type(), privilege.clazz(),
-                        targetIdent, privilege.grantor()));
-                    privilege.setIdent(targetIdent);
+                    ident = targetIdent;
                     affectedPrivileges++;
                 }
+                privileges.add(new Privilege(privilege.state(), privilege.type(), privilege.clazz(),
```

Comment:
Is it worth cloning ALL table privileges? The privilege object is immutable. I think we should just create a new privilege object for privilege we transfer